gax/src/main/java/com/google/api/gax/rpc/ApiCallContext.java

```diff
@@ -60,19 +60,38 @@
   ApiCallContext withTransportChannel(TransportChannel channel);
 
   /**
-   * Returns a new ApiCallContext with the given timeout set.
+   * Returns a new ApiCallContext with the given RPC timeout set.
    *
    * <p>This sets the maximum amount of time a single unary RPC attempt can take. If retries are
-   * enabled, then this can take much longer. Unlike a deadline, timeouts are relative durations
-   * that are measure from the beginning of each RPC attempt. Please note that this will limit the
-   * duration of a server streaming RPC as well.
+   * enabled, then this can take much longer. Furthermore, the timeout calculated by the {@link
+   * com.google.api.gax.retrying.RetrySettings} may take precedence over any value provided here.
```

Comment:
I updated this comment because today the AttemptCallable [overrides] this ApiCallContext timeout value with that of the RPC timeout calculated by the RetryAlgorithm TimedAttemptSettings. 

Perhaps we should change AttemptCallable to respect the ApiCallContext.withTimeout if set by the user? This could have odd behavior, though, where both this ApiCallContext.withTimeout and ApiCallContext.withOverallTimeout are set, the RetryAlgorithm could potentially run beyond the `overallTimeout`.

[overrides]: https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/rpc/AttemptCallable.java#L72-L75